code/game/objects/items/airbubble.dm

```diff
@@ -0,0 +1,339 @@
+	else
+		if(contains_body > 0)
+			icon_state = "bodybag_closed1"
+		else
+			icon_state = icon_closed
+
+/obj/structure/closet/air_bubble/proc/process_tank_give_air()
+	if(internal_tank)
+		var/datum/gas_mixture/tank_air = internal_tank.return_air()
+
+		var/release_pressure = internal_tank_valve
+		var/inside_pressure = inside_air.return_pressure()
+		var/pressure_delta = min(release_pressure - inside_pressure, (tank_air.return_pressure() - inside_pressure)/2)
+		var/transfer_moles = 0
+
+		if(pressure_delta > 0) //inside pressure lower than release pressure
+			if(tank_air.temperature > 0)
+				transfer_moles = pressure_delta*inside_air.volume/(inside_air.temperature * R_IDEAL_GAS_EQUATION)
+				var/datum/gas_mixture/removed = tank_air.remove(transfer_moles)
+				inside_air.merge(removed)
+
+		else if(pressure_delta < 0) //inside pressure higher than release pressure
+			var/datum/gas_mixture/t_air = get_turf_air()
+			pressure_delta = inside_pressure - release_pressure
+
+			if(t_air)
+				pressure_delta = min(inside_pressure - t_air.return_pressure(), pressure_delta)
+			if(pressure_delta > 0) //if location pressure is lower than inside pressure
+				transfer_moles = pressure_delta*inside_air.volume/(inside_air.temperature * R_IDEAL_GAS_EQUATION)
+
+				var/datum/gas_mixture/removed = inside_air.remove(transfer_moles)
+				if(t_air)
+					t_air.merge(removed)
+				else //just delete the inside gas, we're in space or some shit
+					qdel(removed)
+
+/obj/structure/closet/air_bubble/proc/process_preserve_temp()
+	if (inside_air && inside_air.volume > 0)
+		var/delta = inside_air.temperature - T20C
+		inside_air.temperature -= max(-10, min(10, round(delta/4,0.1)))
+
+/////////////////////////////////////////
+//////// airbubble process() helpers ////
+/////////////////////////////////////////
+
+/obj/structure/closet/air_bubble/proc/stop_process(process)
+	current_processes &= ~process
+
+/obj/structure/closet/air_bubble/proc/start_process(process)
+	current_processes |= process
+
+/obj/structure/closet/air_bubble/remove_air(amount)
+	if(use_internal_tank)
+		return inside_air.remove(amount)
+	else
+		var/turf/T = get_turf(src)
+		if(T)
+			return T.remove_air(amount)
+	return
+
+/obj/structure/closet/air_bubble/return_air()
+	if(use_internal_tank)
+		return inside_air
+	return get_turf_air()
+
+/obj/structure/closet/air_bubble/proc/get_turf_air()
+	var/turf/T = get_turf(src)
+	if(T)
+		. = T.return_air()
+	return
+
+/obj/structure/closet/air_bubble/proc/return_pressure()
+	. = 0
+	if(use_internal_tank)
+		. =  inside_air.return_pressure()
+	else
+		var/datum/gas_mixture/t_air = get_turf_air()
+		if(t_air)
+			. = t_air.return_pressure()
+	return
+
+
+/obj/structure/closet/air_bubble/proc/return_temperature()
+	. = 0
+	if(use_internal_tank)
+		. = inside_air.temperature
+	else
+		var/datum/gas_mixture/t_air = get_turf_air()
+		if(t_air)
+			. = t_air.temperature
+	return
+
+/obj/structure/closet/air_bubble/process()
+	var/static/max_ticks = 16
```

Comment:
Why isn't this a constant or define?

Reply:
Idk, It was borrowed from Mech's code, which doesn't have it.